sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkGetFunctionsOperation.scala

```diff
@@ -80,8 +83,20 @@ private[hive] class SparkGetFunctionsOperation(
       parentSession.getUsername)
 
     try {
+      val separateDisplaySystemFunctions =
+        sqlContext.conf.getConf(SQLConf.THRIFTSERVER_SEPARATE_DISPLAY_SYSTEM_FUNCTION)
+      var matchedBuiltInFunctions = if (separateDisplaySystemFunctions && functionPattern == "*"
+        && matchingDbs.nonEmpty) {
+        FunctionRegistry.functionSet ++ TableFunctionRegistry.functionSet
+      } else {
+        Set.empty[FunctionIdentifier]
+      }
       matchingDbs.foreach { db =>
         catalog.listFunctions(db, functionPattern).foreach {
+          case (funcIdentifier, "SYSTEM") if separateDisplaySystemFunctions =>
```

Comment:
Please declare a constant variable and use it.

Reply:
Done